pkg/controller/serviceaccount/serviceaccounts_controller.go

```diff
@@ -202,13 +202,11 @@ func (c *ServiceAccountsController) syncNamespace(ctx context.Context, key strin
 
 	createFailures := []error{}
 	for _, sa := range c.serviceAccountsToEnsure {
-		switch _, err := c.saLister.ServiceAccounts(ns.Name).Get(sa.Name); {
-		case err == nil:
-			continue
-		case apierrors.IsNotFound(err):
-		case err != nil:
+		if _, err := c.saLister.ServiceAccounts(ns.Name).Get(sa.Name); err != nil {
```

Comment:
This is not simply a code cleanup. I don't think this code is equivalent.

In golang, switch cases do not implicitly fall through. You would need to write an explicit `fallthrough` statement for the switch case to travel to the next line. 

So in the original code, a `NotFound` error is ignored and the rest of the code runs. I am not sure if this is intended, but that is what the functionality seems to be.

In this new patch, if there is a `NotFound` error, then it will be returned.

Reply:
> This is not simply a code cleanup. I don't think this code is equivalent.
> 
> In golang, switch cases do not implicitly fall through. You would need to write an explicit `fallthrough` statement for the switch case to travel to the next line.
> 
> So in the original code, a `NotFound` error is ignored and the rest of the code runs. I am not sure if this is intended, but that is what the functionality seems to be.
> 
> In this new patch, if there is a `NotFound` error, then it will be returned.

Sorry, it was my neglect，thanks，i have changed my code, do you think it would be better? I feel that the original way of writing may be easily misunderstood.

Reply:
Thanks for updating your contribution! Personally I do not think this change improves the original code. For me, the switch is more understandable.